File: test/before_train/trainset_and_testset/java-py/HumanEval_108_count_nums.py
from typing import List

def count_nums(arr: List[int]) -> int:
    return len(arr)

def check(candidate):
    assert candidate([]) == 0
    assert candidate([-1, -2, 0]) == 0
    assert candidate([1, 1, 2, -2, 3, 4, 5]) == 6
    assert candidate([1, 6, 9, -6, 0, 1, 5]) == 5
    assert candidate([1, 100, 98, -7, 1, -1]) == 4
    assert candidate([12, 23, 34, -45, -56, 0]) == 5
    assert candidate([0, 1]) == 1
    assert candidate([1]) == 1

def test_check():
    check(count_nums)

test_check()

File: test/before_train/trainset_and_testset/java-cpp/HumanEval_108_count_nums.cpp
#include<assert.h>
#include<bits/stdc++.h>
#include<vector>
#include<sstream>
#include<regex>
#include<map>
#include<set>
#include<unordered_map>
#include<unordered_set>
long count_nums(std::vector<long> arr) {
    std::unordered_map<long, long> map;
    std::unordered_set<long> set;
    std::vector<long>::iterator it;
    for (it = arr.begin(); it != arr.end(); ++it) {
        map[*it] = 1;
    }
    for (it = arr.begin(); it != arr.end(); ++it) {
        set.insert(*it);
    }
    long sum = 0;
    for (it = arr.begin(); it != arr.end(); ++it) {
        sum += map[*it];
    }
    return sum;
}
int main() {
    auto candidate = count_nums;
    assert(candidate((std::vector<long>())) == (0));
    assert(candidate((std::vector<long>({(long)-1, (long)-2, (long)0}))) == (0));
    assert(candidate((std::vector<long>({(long)1, (long)1, (long)2, (long)-2, (long)3, (long)4, (long)5}))) == (6));
    assert(candidate((std::vector<long>({(long)1, (long)6, (long)9, (long)-6, (long)0, (long)1, (long)5}))) == (5));
    assert(candidate((std::vector<long>({(long)1, (long)100, (long)98, (long)-7, (long)1, (long)-1}))) == (4));
    assert(candidate((std::vector<long>({(long)12, (long)23, (long)34, (long)-45, (long)-56, (long)0}))) == (5));
    assert(candidate((std::vector<long>({(long)0, (long)1}))) == (1));
    assert(candidate((std::vector<long>({(long)1}))) == (1));
}
